apl-core/src/main/java/com/apollocurrency/aplwallet/apl/core/db/BlockDaoImpl.java

```diff
@@ -350,14 +315,16 @@ public int getBlockCount(long accountId) {
                 rs.next();
                 return rs.getInt(1);
             }
-        } catch (SQLException e) {
+        }
+        catch (SQLException e) {
             throw new RuntimeException(e.toString(), e);
         }
     }
 
     @Override
-    public List<Long> getBlockIdsAfter(long blockId, int limit, List<Long> result) {
+    public List<Long> getBlockIdsAfter(long blockId, int limit) {
         // Search the database
+        List<Long> result = new ArrayList<>(limit);
```

Comment:
Default DataSource for main db is fetched here in all cases. Is it correct for now?

Reply:
Yes, its correct. See getBlockIdsAfter in BlockchainImpl. We extract block ids from index and then forward further search to current db, if required.